core/scripts/vrfv2/superscripts/main.go

```diff
@@ -0,0 +1,257 @@
+			vrf_coordinator_v2.VRFCoordinatorV2FeeConfig{
+				FulfillmentFlatFeeLinkPPMTier1: uint32(500),
+				FulfillmentFlatFeeLinkPPMTier2: uint32(500),
+				FulfillmentFlatFeeLinkPPMTier3: uint32(500),
+				FulfillmentFlatFeeLinkPPMTier4: uint32(500),
+				FulfillmentFlatFeeLinkPPMTier5: uint32(500),
+				ReqsForTier2:                   big.NewInt(0),
+				ReqsForTier3:                   big.NewInt(0),
+				ReqsForTier4:                   big.NewInt(0),
+				ReqsForTier5:                   big.NewInt(0),
+			},
+		)
+		helpers.PanicErr(err)
+		confirmTXMined(context.Background(), ec, tx, chainID)
+
+		fmt.Println("\nConfig set, getting current config from deployed contract...")
+		cfg, err := coordinator.GetConfig(nil)
+		helpers.PanicErr(err)
+		feeConfig, err := coordinator.GetFeeConfig(nil)
+		helpers.PanicErr(err)
+		fmt.Printf("Config: %+v\n", cfg)
+		fmt.Printf("Fee config: %+v\n", feeConfig)
+
+		fmt.Println("\nRegistering proving key...")
+		if strings.HasPrefix(*registerKeyUncompressedPubKey, "0x") {
+			*registerKeyUncompressedPubKey = strings.Replace(*registerKeyUncompressedPubKey, "0x", "04", 1)
+		}
+		pubBytes, err := hex.DecodeString(*registerKeyUncompressedPubKey)
+		helpers.PanicErr(err)
+		pk, err := crypto.UnmarshalPubkey(pubBytes)
+		helpers.PanicErr(err)
+		tx, err = coordinator.RegisterProvingKey(owner,
+			common.HexToAddress(*registerKeyOracleAddress),
+			[2]*big.Int{pk.X, pk.Y})
+		helpers.PanicErr(err)
+		confirmTXMined(
+			context.Background(),
+			ec,
+			tx,
+			chainID,
+			fmt.Sprintf("Uncompressed public key: %s,", *registerKeyUncompressedPubKey),
+			fmt.Sprintf("Oracle address: %s,", *registerKeyOracleAddress),
+		)
+
+		fmt.Println("\nProving key registered, getting proving key hashes from deployed contract...")
+		_, _, s_provingKeyHashes, err := coordinator.GetRequestConfig(nil)
+		helpers.PanicErr(err)
+		fmt.Printf("Hashes: %+v\n", s_provingKeyHashes)
+
+		// Deploy consumer and subscribe.
+		fmt.Println("\nDeploying consumer and subscribing...")
+		keyHashBytes := common.HexToHash(*keyHash)
+		_, tx, _, err = vrf_single_consumer_example.DeployVRFSingleConsumerExample(
+			owner,
+			ec,
+			coordinatorAddress,
+			common.HexToAddress(*linkAddress),
+			uint32(1000000), // gas callback
+			uint16(5),       // confs
+			uint32(1),       // words
+			keyHashBytes)
+		helpers.PanicErr(err)
+		consumerAddress := confirmContractDeployed(context.Background(), ec, tx, chainID)
+		subId := uint64(1)
+
+		fmt.Println("\nFunding subscription...")
+		b, err := utils.GenericEncode([]string{"uint64"}, subId)
+		helpers.PanicErr(err)
+		linkToken, err := link_token_interface.NewLinkToken(common.HexToAddress(*linkAddress), ec)
+		helpers.PanicErr(err)
+		tx, err = linkToken.TransferAndCall(owner, coordinator.Address(), subscriptionBalance, b)
+		helpers.PanicErr(err)
+		confirmTXMined(context.Background(), ec, tx, chainID)
```

Comment:
Can you add some context to these so we know what transaction it is?

Reply:
Yeah I can add an extra parameter or two to the log.